src/components/LogRow/BaseRow/SharingMenu/index.tsx

```diff
@@ -0,0 +1,147 @@
+    if (startingLine === undefined || endingLine === undefined) return;
+    // Create an array of line numbers that represent the range in selectedLines
+    const lineNumbers = getLinesInProcessedLogLinesFromSelectedLines(
+      processedLogLines,
+      selectedLines
+    );
+
+    await copyToClipboard(getJiraFormat(lineNumbers, getLine));
+    setOpen(false);
+    sendEvent({
+      name: "Copied Share Lines To Clipboard",
+    });
+    dispatchToast.success(`Copied ${lineNumbers.length} lines to clipboard`);
+  };
+
+  const handleOnlySearchOnRange = () => {
+    const { endingLine, startingLine } = selectedLines;
+    if (startingLine === undefined || endingLine === undefined) return;
+    setParams({
+      ...params,
+      [QueryParams.LowerRange]: startingLine,
+      [QueryParams.UpperRange]: endingLine,
+    });
+    setOpen(false);
+    sendEvent({
+      name: "Applied Range Limit",
+    });
+  };
+
+  const handleShareLinkToSelectedLines = async () => {
+    const { startingLine } = selectedLines;
+    if (startingLine === undefined) return;
+    // Take the current URL and add the shareLine query param
+    const url = new URL(window.location.href);
+    url.searchParams.set(QueryParams.ShareLine, startingLine.toString());
+
+    await copyToClipboard(url.toString());
+    setOpen(false);
+    sendEvent({ name: "Copied Share Link" });
+    dispatchToast.success("Copied link to clipboard");
+  };
+
+  const lineCount =
+    selectedLines.endingLine === undefined ||
+    selectedLines.startingLine === undefined
+      ? 1
+      : selectedLines.endingLine - selectedLines.startingLine + 1;
+
+  return (
+    <StyledMenu
+      align="right"
+      data-cy="sharing-menu"
+      open={open}
+      setOpen={setMenuOpen}
+      trigger={
+        <MenuIcon
+          aria-label="Expand share menu"
+          data-cy="sharing-menu-button"
+          onClick={setMenuOpen}
+        >
+          <Icon glyph="Ellipsis" />
+        </MenuIcon>
+      }
+    >
+      <MenuItem
+        glyph={<Icon glyph="Copy" />}
+        onClick={handleCopySelectedLines}
+        title={`Copy the selected ${pluralize(
+          "line",
+          lineCount
+        )} to your clipboard with JIRA formatting.`}
+      >
+        Copy selected {pluralize("line", lineCount)}
+      </MenuItem>
+      {!isUploadedLog && (
+        <MenuItem
+          glyph={<Icon glyph="Export" />}
+          onClick={handleShareLinkToSelectedLines}
+          title={`Copy a link to ${pluralize("this", lineCount)} ${pluralize(
+            "line",
+            lineCount
+          )}.`}
+        >
+          Share link to selected {pluralize("line", lineCount)}
+        </MenuItem>
+      )}
+      <MenuItem
+        disabled={lineCount === 1}
+        glyph={<Icon glyph="MagnifyingGlass" />}
+        onClick={handleOnlySearchOnRange}
+        title="Limit the range Parsley will search to only these lines."
+      >
+        Only search on range
```

Comment:
This is super extra, but it would be cool if clicking this button caused some kind of pulse or animation on the "Details" button to indicate that the range setting is there. If you like the idea could be worth just opening a ticket for.

Reply:
I think this is actually a great idea! I opened a follow-up ticket for it to do it this sprint since this pr is already pretty large. https://jira.mongodb.org/browse/DEVPROD-1647